File: laravel/Libreria/app/Http/Controllers/AuthorController.php
<?php

namespace App\Http\Controllers;

use App\Models\Author;
use Illuminate\Http\Request;

class AuthorController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        $authors = Author::all();
        return view("authors.index",compact("authors"));
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("authors.create");
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Author::create($request->all());
        return redirect('/');
    }

    /**
     * Display the specified resource.
     */
    public function show(Author $author)
    {
        return view("authors.show", compact("author"));
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Author $author)
    {
        return view("authors.edit",compact('author'));
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Author $author)
    {
        $author->update($request->all());
        return redirect('/authors');
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Author $author)
    {
        $author->delete();
        return redirect('/authors');
    }
}

File: javaSpring/Libreria/src/main/java/com/example/Libreria/controllers/AuthorController.java
package com.example.Libreria.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;

import com.example.Libreria.models.Author;
import com.example.Libreria.repositories.AuthorRepository;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;



@Controller
public class AuthorController {
    

    final AuthorRepository repo;
    

    public AuthorController(AuthorRepository repo){
        this.repo=repo;
    }

    @GetMapping("")
    public String home() {
        return "index";
    }
    

    @GetMapping("/authors")
    public String index(Model model) {
        model.addAttribute("authors",repo.findAll());
        return "authors/index";
    }

    @GetMapping("/authors/create")
    public String create() {
        return "/authors/create";
    }

    @PostMapping("/authors/store")
    public String store(Author a) {
        repo.save(a);
        return "redirect:/authors";
    }

    @GetMapping("/authors/show")
    public String show(Model model,Long id){
        model.addAttribute("author",repo.findById(id).orElse(null));
        return "/authors/show";
    }

    @GetMapping("/authors/edit")
    public String edit(Model model,Long id){
        model.addAttribute("author",repo.findById(id).orElse(null));
        return "/authors/edit";
    }
    
    @PostMapping("/authors/update")
    public String update(Author author){
        repo.save(author);
        return "redirect:/authors";
    }

    @PostMapping("/authors/delete")
    public String delete(Long id){
        repo.deleteById(id);
        return "redirect:/authors";
    }

}
